### PR TITLE
Introduce as-shot to later D65 workflow

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
Replaces #15461 so maybe read discussions there 

Please note that temperature module version is increased in the PR

The current modern chroma correction workflow assumes "camera reference point" (nomally D65) has the best available data for temperature coeffs as preliminary for all modules in the pixelpipe before colorin. There we change to working profile and continue work for best visual quality until the "color calibration" module.

There is one drawback with this approach. Some modules in the pipe would like to have "perfect white balance" correction - the rgb channels all have the same value for any greytone.

Examples:
1. Some hightlights reconstruction algos take the other channels or surrounding data into account and modify data "towards white".
2. raw chromatic aberration correction also iterates from channel differences. Good "white is white" coeffs help significantly here.
3. Some demosaicers also have slightly improved output on pixelpeeping level.
4. ...

In short - how does it work?

```
was: data(rgb) -> temperature(*D65)     -> ModA -> ModB ...                   ->colorin
now: data(rgb) -> temperature(*as_shot) -> ModA -> ModB ... -> *(D65/as_shot) ->colorin
```

### CODE REVIEW

**Comment by jenshannoschwalm** [comment]
@rawfiner would you want to comment?

**Comment by rawfiner** [comment]
I won't have time to test the PR in the next days unfortunately, but I really think the idea it implements is good :-)

**Comment by yuri1969** [comment]
I'd love to test this PR using the shot from #15121. What is the required module sequence?

Is it enough to just enable the new 5th _white balance module_ mode?

**Comment by jenshannoschwalm** [comment]
Yes. And iirc that image was about chromatic aberration. So enable raw chromatic aberration also. Then you can swap the 4th and 5th bulb to see effect.

**Comment by yuri1969** [comment]
@jenshannoschwalm Strange, I can see no difference between 4th and 5th bulbs having the _raw chromatic aberration_ module with _avoid colorshift_ enabled.

The green tint is still present:
20231029-201023_snap
20231029-201113_snap

**Comment by jenshannoschwalm** [comment]
I think i once had the raw image. could you re-share/post please?

**Comment by yuri1969** [comment]
Sure, here it is:
[DSC09369.ARW.txt](https://github.com/darktable-org/darktable/files/13220766/DSC09369.ARW.txt)

**Comment by jenshannoschwalm** [comment]
Thanks. Indeed i see the tinting, it's definitely not related to - and can't be fixed by - this pr though. Just check with temperature "as-shot" and disabled color calibration. Will check

**Comment by jenshannoschwalm** [comment]
Latest commit fixed a bug with preset not kept correctly

**Comment by TurboGit** [comment]
@jenshannoschwalm : As a first comment I'll say that it looks like another option to the white balance module which is probably hard to document (so hard to understand). I think that I'll need some examples (can you share one or more RAW) where it makes a difference using the new mode. Maybe (probably) me but I have hard time to fully understand the gain, yes I do understand what it does technically... But what it brings in terms of value? I have played with some of my RAW and I've yet to see a difference using chromatic aberration or highlight reconstruction. Or maybe my expectations are wrong? So I need some more explanation here. TIA.

**Comment by jenshannoschwalm** [comment]
> So I need some more explanation here. TIA.

There is still something wrong i'll have to fix before discussing intention & details.

**Comment by jenshannoschwalm** [comment]
If testing this please keep in mind
1. You will observe differences between new mode and modern-D65 (the bulb) if there are a) modules enabled that change output "relative to white" and b) as-shot coefficients are no-so-close to D65
2. For highlights you can expect blown out areas having better estimations (more to white)
3. For raw ca correction there are often also slightly better data
4. very subtle differences for LMMSE or RCD demosaicers

Here is one example file provided by @IainMF while working on highlights module.
[Tall unicycle.zip](https://github.com/darktable-org/darktable/files/13258722/Tall.unicycle.zip)
[Tall unicycle.NEF.xmp.txt](https://github.com/darktable-org/darktable/files/13258726/Tall.unicycle.NEF.xmp.txt)

**Comment by jenshannoschwalm** [comment]
Release note:
Introduced a new mode for 'white balance' module: "as shot to reference" as an alternative to "camera reference". By using this preset, all modules in the pixelpipe until 'input color profile' will use the white balance coefficients found by "as-shot" providing better coeffs in some cases. 'input color profile' takes care of this by correcting input to "camera reference"

**Comment by jenshannoschwalm** [comment]
ok, i can confirm that. will mark as drafted as long as i am not completely sure if this helps in a significant number of images or i can do somewhat better algorithm-wise

**Comment by TurboGit** [comment]
> 2\. For highlights you can expect blown out areas having better estimations (more to white)

I'm not seeing that on my side, more often I have more redish into the blown highlight when activating the new mode. See RAW + xmp to reproduce:

https://drive.google.com/file/d/1my7OEnZf_7pIgPXULNQHppzl_aVv7hWt/view?usp=sharing

The xmp (just change from D65 to new late D65):

[raw_ch7_part1_pap3.arw.xmp.txt](https://github.com/darktable-org/darktable/files/13259051/raw_ch7_part1_pap3.arw.xmp.txt)

**Comment by jenshannoschwalm** [comment]
closing for now ...

**Comment by rawfiner** [comment]
As noted by @wpferguson in previous PR, this PR give visible improvements on denoise profile.

For instance, with this image (credits jpg54 from darktablefr):
https://drive.google.com/file/d/1aca3bBqS28yJSTsEcXkyli3MQdPbCM0d/view?usp=sharing

Color fidelity is much better with late D65 (this PR), when using denoise profile in wavelets. Here, I use the chroma preset.
Unwanted blue cast in dark area (without this PR):
Capture d’écran du 2023-11-07 17-48-23
No color cast and better details (with this PR):
Capture d’écran du 2023-11-07 17-48-19

This is because Y0U0V0 decomposition and bias correction work better with white balanced images.
The improvement is visible for many indoor shots, as long as the white balance is very different from the neutral one.

Another example (crop of a picture I cannot share unfortunately):
The blue cast in the hair without this PR is quite disturbing:
Capture d’écran du 2023-11-07 17-53-46
With this PR, no issue anymore:
Capture d’écran du 2023-11-07 17-53-56

**Comment by jenshannoschwalm** [comment]
Yes, i fully agree with your analysis. Also true for sure with raw chroma correction.
Unfortunately for highlights correction this is a twofold story
1. correction is certainly better with new algo
2. color cast is often worse. I will think about this again - is there a solution - and reopen if i got it :-) For sure closing this is only "for now"
